[Bug] - Conversion of old boolean values to new ExporterEnabledState does not work when in quotes

**Describe the bug**
When the boolean for the enabled property is written in quotes, it is treated as a String by Spring and therefore can not be converted using the BooleanToExporterEnabledStateConverter

**To Reproduce**
Steps to reproduce the behavior:
1. Set an enabled property to `"false"`
2. Start any core test that loads properties
3. See error in logs

**Expected behavior**
Convert the value to equivalent ExporterEnabledState

**Additional information (screenshots, stack traces, etc.)**
Caused by: org.springframework.core.convert.ConversionFailedException: Failed to convert from type [java.lang.String] to type [rocks.inspectit.ocelot.config.model.exporters.ExporterEnabledState] for value 'false'; nested exception is java.lang.IllegalArgumentException: No enum constant rocks.inspectit.ocelot.config.model.exporters.ExporterEnabledState.false

--- a/inspectit-ocelot-config/src/main/java/rocks/inspectit/ocelot/config/conversion/InspectitConfigConversionService.java
+++ b/inspectit-ocelot-config/src/main/java/rocks/inspectit/ocelot/config/conversion/InspectitConfigConversionService.java
@@ -18,5 +18,6 @@ public class InspectitConfigConversionService extends ApplicationConversionServi
         addConverter(new StringToMetricRecordingSettingsConverter());
         addConverter(new NumberToMetricRecordingSettingsConverter());
         addConverter(new BooleanToExporterEnabledStateConverter());
+        addConverter(new StringToExporterEnabledStateConverter());
     }
 }

--- a/inspectit-ocelot-config/src/main/java/rocks/inspectit/ocelot/config/conversion/StringToExporterEnabledStateConverter.java
+++ b/inspectit-ocelot-config/src/main/java/rocks/inspectit/ocelot/config/conversion/StringToExporterEnabledStateConverter.java
@@ -1,0 +1,36 @@
+package rocks.inspectit.ocelot.config.conversion;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.convert.converter.Converter;
+import rocks.inspectit.ocelot.config.model.exporters.ExporterEnabledState;
+
+/**
+ * Before version 1.15, the enabled property of exporters was a Boolean, now it is a {@link ExporterEnabledState}.
+ * To support not-yet-updated configurations, this {@link BooleanToExporterEnabledStateConverter} converts Boolean values
+ * to their equivalent new values, i.e. 'true' to 'IF_CONFIGURED' and 'false' to 'DISABLED'.
+ * However, in Spring Booleans can be defined using Strings, i.e. quoted Boolean values like '"false"', as well, so this
+ * converter is additionally needed to cover that case. However, because the correct new values are also Strings at first,
+ * that are then converted to ExporterEnabledStates, this class also needs to handle that correct usage.
+ * The class is deprecated, since the Boolean conversion functionality will be removed and configurations using the old style
+ * made invalid in a future release of InspectIT Ocelot, and the conversion for correct values will without this
+ * Converter work out-of-the-box in Spring.
+ */
+@Slf4j
+@Deprecated
+public class StringToExporterEnabledStateConverter implements Converter<String, ExporterEnabledState> {
+
+    @Override
+    public ExporterEnabledState convert(String source) {
+        if (source.equalsIgnoreCase("true")) {
+            // log deprecation warn
+            log.warn("You are using the deprecated Boolean-based style with quotes to define whether an exporter is enabled. This style will be invalid in future releases of InspectIT Ocelot.");
+            return ExporterEnabledState.IF_CONFIGURED;
+        } else if (source.equalsIgnoreCase("false")) {
+            // log deprecation warn
+            log.warn("You are using the deprecated Boolean-based style with quotes to define whether an exporter is enabled. This style will be invalid in future releases of InspectIT Ocelot.");
+            return ExporterEnabledState.DISABLED;
+        } else {
+            return ExporterEnabledState.valueOf(source);
+        }
+    }
+}

--- a/inspectit-ocelot-config/src/main/java/rocks/inspectit/ocelot/config/conversion/StringToExporterEnabledStateConverter.java
+++ b/inspectit-ocelot-config/src/main/java/rocks/inspectit/ocelot/config/conversion/StringToExporterEnabledStateConverter.java
@@ -21,14 +21,10 @@ public class StringToExporterEnabledStateConverter implements Converter<String, 
 
     @Override
     public ExporterEnabledState convert(String source) {
-        if (source.equalsIgnoreCase("true")) {
+        if (source.equalsIgnoreCase("true") || source.equalsIgnoreCase("false")) {
             // log deprecation warn
             log.warn("You are using the deprecated Boolean-based style with quotes to define whether an exporter is enabled. This style will be invalid in future releases of InspectIT Ocelot.");
-            return ExporterEnabledState.IF_CONFIGURED;
-        } else if (source.equalsIgnoreCase("false")) {
-            // log deprecation warn
-            log.warn("You are using the deprecated Boolean-based style with quotes to define whether an exporter is enabled. This style will be invalid in future releases of InspectIT Ocelot.");
-            return ExporterEnabledState.DISABLED;
+            return source.equalsIgnoreCase("true") ? ExporterEnabledState.IF_CONFIGURED : ExporterEnabledState.DISABLED;
         } else {
             return ExporterEnabledState.valueOf(source);
         }

